lms/templates/navigation.html

```diff
@@ -21,6 +20,26 @@
 ## Provide a hook for themes to inject branding on top.
 <%block name="navigation_top" />
 
+## Add UI Toolkit components if using the Pattern Library
+% if uses_pattern_library:
+    <%block name="js_extra">
+    <%static:require_module module_name="js/header_factory" class_name="HeaderFactory">
+    HeaderFactory();
+    </%static:require_module>
+    </%block>
+% endif
+
+<%
+if uses_pattern_library:
```

Comment:
Sorry, I thought that you and @AlasdairSwan has settled upon an approach. 

My feeling is that it would be more disruptive to try to change the existing CSS as part of this work. I'd rather just consider that to be legacy, and move forward on cleaner markup for new pages. These kind of conditional changes will only be in the header and footer as they are the only elements that are in common between the old and new style pages. Given this, I feel like this is the lesser of two evils, and keeps the scope of this story smaller.

Let me know what you think.